backend/lib/dbHelpers.js

```diff
@@ -47,6 +53,29 @@ module.exports = (client) => {
       cb(null, reply)
     })
   }
+  this.getProductIdsByCategories = Bluebird.promisify(getProductIdsByCategories)
+
+  this.getProductObjsArrByCategories = (categoriesArr, cb) => {
+    this.getProductIdsByCategories(categoriesArr).then(
+      (productIdsArr) => {
+        if (productIdsArr.length === 0) {
+          cb(null, productIdsArr)
+        }
+        else {
+          var productObjsPromises = productIdsArr.map( (productId) => {
+            return this.getProductById(productId)
+          })
+          // console.log(productObjsPromises);
+          Bluebird.all(productObjsPromises).then(
```

Comment:
If you return the result of `Bluebird.all()` here you can call `then` one level up, like this:
```javascript
somePromise.then(function(result) {
  var arrayOfPromises = getArrayOfPromisesSomehow()
  return Bluebird.all(arrayOfPromises)
}).then(function(resultOfBluebirdAll) {
  // ...
})
```
Helps to reduce nesting slightly.